src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2054,7 +2073,15 @@ protected function getMorphs($name, $type, $id)
      */
     public function getMorphClass()
     {
-        return $this->morphClass ?: get_class($this);
+        $morphMap = Relation::morphMap();
+        $class = get_class($this);
+
+        if (! empty($morphMap) &&
+            in_array($class, $morphMap)) {
```

Comment:
because it doesn;t need to be on two lines, and the cs is wrong for being on two lines anyway